gitlab/base.py

```diff
@@ -249,6 +250,11 @@ def total(self) -> int:
         return self._list.total
 
 
+class RequiredOptional(NamedTuple):
+    required: Tuple[str, ...]
+    optional: Tuple[str, ...]
```

Comment:
Would it make sense to add default values (empty tuples) like the example in https://docs.python.org/3/library/typing.html#typing.NamedTuple?

Then people would only explicitly add required / optional args if needed. Might make the classes a bit more concise. I haven't tried though :)

Reply:
We could do that.

I set up a default value at:
https://github.com/python-gitlab/python-gitlab/pull/1366/files#diff-4d0a1fd6f7611cc6ced65bac48f9cbe658039289f350e2c5d7881674ff61dd9eR267-R272

But that doesn't help when people are setting the value.

Let me investigate it and see how it works.

Reply:
@nejch  I did what you suggested and updated all the `v4/objects/*.py` based on having default values.